docs/plugins/api.asciidoc

```diff
@@ -4,18 +4,6 @@
 API extension plugins add new functionality to Elasticsearch by adding new APIs or features, usually to do with search or mapping.
 
 [float]
-=== Core API extension plugins
```

Comment:
Probably a good thing for another PR: maybe we should mention modules here just to explain that the plugin system is powerful. Or something, I dunno. Maybe just a thing for a blog post.